feat: use 4h timeout default rh-advisories and rh-push-to-registry-redhat-io pipelines

creydr opened the pull request:
By default the task timeout in the release pipelines are 2h. This is especially short for larger applications without having [RELEASE-1291](https://issues.redhat.com/browse/RELEASE-1291) in place and running into konflux-ci/release-service/issues/603.
This PR sets the timeout for the tasks in the rh-advisories pipeline to 4h (what we've done for the OpenShift Serverless release). 
Also updated `rh-push-to-registry-redhat-io` as you try to keep those pipelines in sync.

An alternative would be to set the 4h timeout globally for all tasks in https://github.com/redhat-appstudio/infra-deployments

creydr commented:
/cc @ralphbean

mmalina commented:
/ok-to-test

mmalina commented:
> This is especially short for larger applications with having [RELEASE-1291](https://issues.redhat.com/browse/RELEASE-1291) in place.

You mean "without having RELEASE-1291 in place", right?

It's fine with me, let's see what Johnny thinks. But could you please do the same for the push-to-registry-redhat-io pipeline? We try to keep them in sync with the exception of having/not-having the advisory stuff.

creydr commented:
> > This is especially short for larger applications with having [RELEASE-1291](https://issues.redhat.com/browse/RELEASE-1291) in place.
> 
> You mean "without having [RELEASE-1291](https://issues.redhat.com//browse/RELEASE-1291) in place", right?

Oh, yes. Updated the PR description 

> It's fine with me, let's see what Johnny thinks. But could you please do the same for the push-to-registry-redhat-io pipeline? We try to keep them in sync with the exception of having/not-having the advisory stuff.

Updated rh-push-to-registry-redhat-io as well

> Also, you'll need to fix your commit message - see here: https://github.com/konflux-ci/release-service-catalog/blob/development/CONTRIBUTING.md#commit-message-formatting-and-standards

Updated

ralphbean commented:
/ok-to-test

johnbieren commented:
I don't really get doing this for all tasks. Can you explain? For example, I would *never* expect `verify-access-to-resources` to not pass after 5 mins but pass after 200 minutes. Also, gitlint is failing. I think it is due to the capital U in Use in the commit title

davidmogar commented:
Agree with Johnny. Having a general timeout sounds bad to me to be honest. Maybe timeouts have to be raised but they shouldn't be treated in the same way.

mmalina commented:
> I don't really get doing this for all tasks. Can you explain? For example, I would _never_ expect `verify-access-to-resources` to not pass after 5 mins but pass after 200 minutes. Also, gitlint is failing. I think it is due to the capital U in Use in the commit title

The current default is 2 hours. You could say the same today - that 2 hours don't make sense in some cases. Raising all of them is just the easiest way to improve the situation - otherwise it's a lot more work to analyse how much is reasonable for each task. Besides, as was discussed in Slack recently, the main issue is currently that taskruns can be waiting for the PV to be freed up for a long time and this will eat away from the timeout value. So it doesn't matter if a task itself never takes more than 5 minutes - it can still time out if some other task blocks it (uses the shared PV). All of this is meant as a temporary measure until that thing is addressed. Of course in the case of `verify-access-to-resources` you could argue that it runs before any of the time consuming tasks start, so it's not likely that anything else will ever block it, but again, in that case you would have to analyse all the tasks to decide where it makes sense or not.

mmalina commented:
/ok-to-test

johnbieren commented:
> > I don't really get doing this for all tasks. Can you explain? For example, I would _never_ expect `verify-access-to-resources` to not pass after 5 mins but pass after 200 minutes. Also, gitlint is failing. I think it is due to the capital U in Use in the commit title
> 
> The current default is 2 hours. You could say the same today - that 2 hours don't make sense in some cases. Raising all of them is just the easiest way to improve the situation - otherwise it's a lot more work to analyse how much is reasonable for each task. Besides, as was discussed in Slack recently, the main issue is currently that taskruns can be waiting for the PV to be freed up for a long time and this will eat away from the timeout value. So it doesn't matter if a task itself never takes more than 5 minutes - it can still time out if some other task blocks it (uses the shared PV). All of this is meant as a temporary measure until that thing is addressed. Of course in the case of `verify-access-to-resources` you could argue that it runs before any of the time consuming tasks start, so it's not likely that anything else will ever block it, but again, in that case you would have to analyse all the tasks to decide where it makes sense or not.

We do not have 2 hour timeouts set for every task in our pipeline definitions. If we did, then the diff would just be changing a 2 to a 4. So I disagree on that. But I do agree that setting a 2 hour timeout for most of the tasks makes no sense. Nowhere in the commit or PR does it say this is a temporary workaround. So, for those reasons, I did not approve.

mmalina commented:
> We do not have 2 hour timeouts set for every task in our pipeline definitions. If we did, then the diff would just be changing a 2 to a 4. So I disagree on that. But I do agree that setting a 2 hour timeout for most of the tasks makes no sense. 

What I meant is that 2 hours is the cluster default, so we currently have 2 hours even for tasks that are not expected to take more than a minute. But I guess that won't change your stance :)

> Nowhere in the commit or PR does it say this is a temporary workaround. So, for those reasons, I did not approve.

I think it's implied by mentioning that this is needed because [RELEASE-1291](https://issues.redhat.com/browse/RELEASE-1291) is not fixed. But I might be wrong.

creydr commented:
> > Nowhere in the commit or PR does it say this is a temporary workaround. So, for those reasons, I did not approve.
> 
> I think it's implied by mentioning that this is needed because [RELEASE-1291](https://issues.redhat.com/browse/RELEASE-1291) is not fixed. But I might be wrong.

Yes, we need this until the volume issue is fixed. Otherwise larger applications will most likely run into the 2h timeouts again. This for sure can be undone after RELEASE-1291 is done and no task is blocked, as another task is still holding the required volume, but IMO there should be a way for bigger applications around this in the meantime.

creydr commented:
> New changes are detected. LGTM label has been removed.

Sorry, needed to adjust the commit message to align with https://github.com/konflux-ci/release-service-catalog/blob/development/CONTRIBUTING.md#commit-message-formatting-and-standards again

johnbieren commented on pipelines/managed/rh-advisories/rh-advisories.yaml:
```diff
@@ -67,6 +67,7 @@ spec:
       value: $(tasks.create-advisory.results.advisory_url)
   tasks:
     - name: verify-access-to-resources
+      timeout: "4h"
```
This task doesn't even use a workspace, I don't think it should have this added

creydr replied:
Updated also for the other tasks which don't have workspaces

johnbieren reviewed (commented):
I think it would be good to add ` # temp workaround until RELEASE-1291 is added` or something to each timeout line being added

creydr committed:
feat: use a default 4h task timeout in rh-advisories pipeline

Explicitly set the timeout for taskruns in the rh-advisories and
rh-push-to-registry-redhat-io pipelines to 4h and thus override the
cluster default of (currently) 2h.
This is especially helpful for larger components which are running
into issues related to RELEASE-1291.

Signed-off-by: Christoph Stäbler <cstabler@redhat.com>

creydr commented:
> I think it would be good to add ` # temp workaround until RELEASE-1291 is added` or something to each timeout line being added

Updated

creydr commented:
Needed to fix the `too few spaces before comment` yaml linter issue

johnbieren commented:
/retest

johnbieren commented:
/retest

johnbieren commented:
@creydr once this passes e2e do you want me to merge this or wait?

johnbieren commented:
/retest

johnbieren commented:
/retest

johnbieren commented:
/retest

johnbieren commented:
/retest

johnbieren commented:
/retest

johnbieren commented:
/retest

creydr commented:
> @creydr once this passes e2e do you want me to merge this or wait?

From my side, no need to wait

johnbieren commented:
/retest

johnbieren commented:
/retest

johnbieren commented:
/retest

johnbieren commented:
/retest

johnbieren commented:
/retest

mmalina commented:
@creydr so it turns out that because your forked repo has a different name, e2e can't run (until https://issues.redhat.com/browse/KFLUXDP-144 is done anyway). Can you try renaming your forked repo to release-service-catalog? I'm not sure what will happen here - you might need to open a new PR as well.

creydr commented:
> @creydr so it turns out that because your forked repo has a different name, e2e can't run (until https://issues.redhat.com/browse/KFLUXDP-144 is done anyway). Can you try renaming your forked repo to release-service-catalog? I'm not sure what will happen here - you might need to open a new PR as well.

Oh ok. I renamed the fork. 
Lets see
/retest

creydr commented:
/retest

mmalina commented:
> @mmalina it's not picking up my `/retest`. So I am either lacking some permissions, or I need to recreate the PR. Can you try a `/retest`?

Thanks. Yeah, I don't think you have permissions. Let's see.

mmalina commented:
/retest

mmalina commented:
@creydr I merged another PR and now there's a conflict, because one task in the rh-advisories pipeline was renamed. I wanted to fix it for you, but I don't have permissions to push to your branch. So either please give me permissions, or you can fix it yourself.

mmalina commented:
/ok-to-test

mmalina commented:
> > @creydr I merged another PR and now there's a conflict, because one task in the rh-advisories pipeline was renamed. I wanted to fix it for you, but I don't have permissions to push to your branch. So either please give me permissions, or you can fix it yourself.
> 
> Strange, I didn't update the permissions by intention. Anyhow, I rebased it

OK, cool. I am not sure right now, but I think there is a checkbox when creating a PR to give push access to maintainers of the target repo or something like that. Not sure if there is a way to change it once the PR is created.

mmalina commented:
Hmm, weird, it seems github is having some issues - several workflows failed to fetch the action. But nothing reported here: https://www.githubstatus.com
Let's retry in a little bit.

johnbieren commented:
/retest

johnbieren commented:
/retest

johnbieren commented:
/retest

johnbieren commented:
/retest

johnbieren commented:
This just passed e2e and has nothing to do with the hot fix stuff we merged since it did. I am going to rebase this and merge it without e2e today

happybhati commented:
### Promoted to stage
